shadowsocks/salt_generator.go

```diff
@@ -64,23 +78,28 @@ var serverSaltLabel = []byte("outline-server-salt")
 // random, but is secretly marked as being issued by the server.
 // This is useful to prevent the server from accepting its own output in a
 // reflection attack.
-func NewServerSaltGenerator(secret string) *ServerSaltGenerator {
+func NewServerSaltGenerator(cipher shadowaead.Cipher, secret string) ServerSaltGenerator {
```

Comment:
Could you pass the salt size instead? Depending on Cipher makes this code less reusable and it's better if this library is lighter and doesn't depend on go-shadowsocks2.

Reply:
OK, I removed the dependency on cipher and reverted to the previous structure.